tests/unit/data/_async/test_client.py

```diff
@@ -3101,119 +3112,105 @@ def resonse_with_result(self, *args, resume_token=None):
         )
 
     @CrossSync.pytest
-    async def test_execute_query(self):
+    async def test_execute_query(self, client, execute_query_mock, prepare_mock):
         values = [
-            self.resonse_with_metadata(),
             self.resonse_with_result("test"),
             self.resonse_with_result(8, resume_token=b"r1"),
             self.resonse_with_result("test2"),
             self.resonse_with_result(9, resume_token=b"r2"),
             self.resonse_with_result("test3"),
             self.resonse_with_result(None, resume_token=b"r3"),
         ]
-        client = self._make_client()
-        with mock.patch.object(
-            client._gapic_client, "execute_query", CrossSync.Mock()
-        ) as execute_query_mock:
-            execute_query_mock.return_value = self._make_gapic_stream(values)
+        execute_query_mock.return_value = self._make_gapic_stream(values)
 
-            result = await client.execute_query(
-                f"SELECT a, b FROM {self.TABLE_NAME}", self.INSTANCE_NAME
-            )
-            results = [r async for r in result]
-            assert results[0]["a"] == "test"
-            assert results[0]["b"] == 8
-            assert results[1]["a"] == "test2"
-            assert results[1]["b"] == 9
-            assert results[2]["a"] == "test3"
-            assert results[2]["b"] is None
-            assert execute_query_mock.call_count == 1
-
-    @CrossSync.pytest
-    async def test_execute_query_with_params(self):
+        result = await client.execute_query(
+            f"SELECT a, b FROM {self.TABLE_NAME}", self.INSTANCE_NAME
+        )
+        results = [r async for r in result]
+        assert results[0]["a"] == "test"
+        assert results[0]["b"] == 8
+        assert results[1]["a"] == "test2"
+        assert results[1]["b"] == 9
+        assert results[2]["a"] == "test3"
+        assert results[2]["b"] is None
+        assert execute_query_mock.call_count == 1
```

Comment:
shouldn't there be an assertion against prepare_mock too?

Reply:
Done